ControllerDriver/networkhandler.cpp

```diff
@@ -83,26 +92,71 @@ bool isConnected() {
   return (wStatus == WL_CONNECTED);
 }
 
-template <typename T>
-void swapEndian(T &val) {
-    union U {
-        T val;
-        std::array<std::uint8_t, sizeof(T)> raw;
-    } src, dst;
-
-    src.val = val;
-    std::reverse_copy(src.raw.begin(), src.raw.end(), dst.raw.begin());
-    val = dst.val;
+template <>
+void swapEndian<uint32_t>(uint32_t &val) {
+    val = htonl(val);
+}
+template <>
+void swapEndian<uint16_t>(uint16_t &val) {
+    val = htons(val);
+}
+template <>
+void swapEndian<uint8_t>(uint8_t &val) {
+    // do nothing.
+}
+template <>
+void swapEndian<int8_t>(int8_t &val) {
+    // do nothing.
 }
 
 template <typename T>
-inline void swapEndianIfNeeded(T &val) {
-  // ifdef to determine byte order? ESP32 and Arduino is little-endian and needs a byte swap for network byte order.
+void writeNumberToBuffer(T val, uint8_t* buf, const uint16_t &pos) {
   swapEndian(val);
+  memcpy(&buf[pos], (uint8_t*) &val, sizeof(T));
```

Comment:
You shouldn't cast `&val` to `uint8_t*` here for two reasons: (1) it's undefined behavior, specifically type punning, and (2) `memcpy` uses void pointers, so the cast is useless anyway.  
```suggestion
  memcpy(&buf[pos], &val, sizeof(T));
```